persistence-jpa/javadsl/src/main/java/com/lightbend/lagom/internal/javadsl/persistence/jpa/JpaSessionImpl.java

```diff
@@ -54,7 +59,12 @@ public JpaSessionImpl(
 
     this.slickDb = slick.db();
     this.factoryCompletionStage = createEntityManagerFactory(actorSystem);
-    lifecycle.addStopHook(this::close);
+
+    Supplier<CompletionStage<Done>> closeSupplier = this::close;
+    coordinatedShutdown.addTask(
+        CoordinatedShutdown.PhaseBeforeActorSystemTerminate(),
+        "shutdown-read-side-jpa",
+        closeSupplier);
```

Comment:
This is the actual fix for JPA. Instead of shutting down the DB early (on `lifecycle.addStopHook`) use a task on a late phase of `coordinatedShutdown` instead.